File: frontend/src/services/eventStatus.ts
export type Tone = 'safe' | 'warn' | 'danger' | 'info'
export type SecurityEventStatus = 'intercepted' | 'suspicious' | 'allowed'

const LEGACY_EVENT_STATUS_MAP: Record<string, SecurityEventStatus> = {
  blocked: 'intercepted',
  pending: 'suspicious',
  closed: 'allowed',
}

export function normalizeEventStatus(status?: string | null): SecurityEventStatus | string {
  const lowered = (status || '').trim().toLowerCase()
  if (lowered in LEGACY_EVENT_STATUS_MAP) {
    return LEGACY_EVENT_STATUS_MAP[lowered]
  }
  if (lowered === 'intercepted' || lowered === 'suspicious' || lowered === 'allowed') {
    return lowered
  }
  return status || ''
}

export function eventStatusTone(status?: string | null): Tone {
  const normalized = normalizeEventStatus(status)
  if (normalized === 'intercepted') return 'safe'
  if (normalized === 'suspicious') return 'warn'
  if (normalized === 'allowed') return 'info'
  return 'info'
}

export function eventStatusLabel(status?: string | null) {
  const normalized = normalizeEventStatus(status)
  if (normalized === 'intercepted') return '已拦截'
  if (normalized === 'suspicious') return '可疑'
  if (normalized === 'allowed') return '已放行'
  return status || '未记录'
}

export function eventStatusRank(status?: string | null) {
  const normalized = normalizeEventStatus(status)
  if (normalized === 'suspicious') return 0
  if (normalized === 'intercepted') return 1
  if (normalized === 'allowed') return 2
  return 3
}

File: backend/app/services/event_status.py
from __future__ import annotations

EVENT_STATUS_INTERCEPTED = "intercepted"
EVENT_STATUS_SUSPICIOUS = "suspicious"
EVENT_STATUS_ALLOWED = "allowed"

LEGACY_EVENT_STATUS_MAP = {
    "blocked": EVENT_STATUS_INTERCEPTED,
    "pending": EVENT_STATUS_SUSPICIOUS,
    "closed": EVENT_STATUS_ALLOWED,
}


def normalize_event_status(status: str | None, fallback: str = EVENT_STATUS_SUSPICIOUS) -> str:
    lowered = (status or "").strip().lower()
    normalized = LEGACY_EVENT_STATUS_MAP.get(lowered, lowered)
    if normalized in {EVENT_STATUS_INTERCEPTED, EVENT_STATUS_SUSPICIOUS, EVENT_STATUS_ALLOWED}:
        return normalized

    fallback_normalized = LEGACY_EVENT_STATUS_MAP.get(str(fallback or "").strip().lower(), str(fallback or "").strip().lower())
    if fallback_normalized in {EVENT_STATUS_INTERCEPTED, EVENT_STATUS_SUSPICIOUS, EVENT_STATUS_ALLOWED}:
        return fallback_normalized
    return EVENT_STATUS_SUSPICIOUS
